src/dos/drive_fat.cpp

```diff
@@ -1056,15 +1081,25 @@ bool fatDrive::FileExists(const char *name) {
 bool fatDrive::FileOpen(DOS_File **file, char *name, uint32_t flags) {
 	direntry fileEntry;
 	uint32_t dirClust, subEntry;
-	if(!getFileDirEntry(name, &fileEntry, &dirClust, &subEntry)) return false;
-	/* TODO: check for read-only flag and requested write access */
+
+	if (!getFileDirEntry(name, &fileEntry, &dirClust, &subEntry)) {
+		DOS_SetError(DOSERR_FILE_NOT_FOUND);
+		return false;
+	}
+
+	bool is_read_only = (readonly || (fileEntry.attrib & DOS_ATTR_READ_ONLY));
+	if (is_read_only && (flags & 0xf) != OPEN_READ) {
```

Comment:
It's the `(flags & 0xf) != OPEN_READ` which is not very obvious what it's testing/checking (beside the literal logic itself).

Can you lift it to a self-documenting boolean, too?